### PR TITLE
[PH19] Implement The Cinematic Phoenix

### PR DESCRIPTION
**Pull request by PurpleCrowbar**
Implements the Heroes of the Realm card [[The Cinematic Phoenix]].

### CODE REVIEW

**Comment by PurpleCrowbar** [comment]
[[The Cinematic Phoenix]]

**Comment by PurpleCrowbar** [comment]
[This card.](https://scryfall.com/card/ph19/6/the-cinematic-phoenix) Presumably doesn't get caught by the bot because it's not legal in constructed play.

File: Mage.Sets/src/mage/cards/t/TheCinematicPhoenix.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        List<Permanent> tapped = (List<Permanent>) getValue("tappedPermanents");
+        if (tapped == null || tapped.isEmpty()) {
+            return false;
+        }
+        if (tapped.stream().allMatch(permanent -> permanent.isLegendary(game))) {
+            game.getPlayer(source.getControllerId()).won(game);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/TheCinematicPhoenix.java:104]: null check

**Review by xenohedron — COMMENTED** [review]
rest looks good

File: Mage.Sets/src/mage/cards/t/TheCinematicPhoenix.java (hunk 1 above)
Comment (PurpleCrowbar) [reply Mage.Sets/src/mage/cards/t/TheCinematicPhoenix.java]: Fixed

**Pull request merged**